<launch>
  	<param name="servo_serial_port" value="/dev/ttyUSB3" />
  	<param name="servo_baud_rate" value="115200" />
    
    <node pkg="motor_communication" type="motor_control_node" name="motor_control_node" output="screen">
        <param name="motor_serial_port" value="/dev/ttyUSB2" />
        <param name="motor_baud_rate" value="9600" />
        <remap from="motor_control/speed" to="manual_control/speed"/>
        <remap from="motor_control/stop_start" to="manual_control/stop_start"/>   
    </node> 

    
	<node pkg="manual_control" type="manual_control_node" name="manual_control" output="screen"/>	
<arg name="realsense_camera_type" default="Intel(R) RealSense(TM) 3D Camer" />
    
    <arg name="camera" default="camera" />
    <arg name="rgb_frame_id" default="$(arg camera)_rgb_optical_frame" />
    <arg name="depth_frame_id" default="$(arg camera)_depth_optical_frame" />

	<arg name="rgb_frame_w" default="1280" />
	<arg name="rgb_frame_h" default="720" />

    <arg name="depth_unit" default="33.1" />
    <arg name="depth_scale" default="0.001" />
    
    <arg name="depth_fx" default="463.888885" />
    <arg name="depth_fy" default="463.888885" />
    
    <arg name="depth_cx" default="320.0" />
    <arg name="depth_cy" default="240.0" />
    
    <arg name="depth_uv_enable_min" default="0" />
    <arg name="depth_uv_enable_max" default="800" />
    
    <arg name="topic_depth_points_id" default="$(arg camera)/depth/points" />
    <arg name="topic_depth_registered_points_id" default="$(arg camera)/depth_registered/points" />
    
    <arg name="topic_image_rgb_raw_id" default="$(arg camera)/rgb/image_raw" />
    <arg name="topic_image_depth_raw_id" default="$(arg camera)/depth/image_raw" />

    <arg name="topic_image_infrared_raw_id" default="$(arg camera)/ir/image_raw" />
    
    <!-- debug depth unit -->
    <arg name="debug_depth_unit" default="false" />

	<!-- camera calibration file url -->
    <arg name="rgb_camera_info_url" default="" />
    <arg name="ir_camera_info_url" default="" />

    <node name="realsense_node" pkg="realsense_camera" type="realsense_camera_node" output="screen">
    	<param name="realsense_camera_type" value="$(arg realsense_camera_type)" />
    	
    	<param name="rgb_frame_w" value="$(arg rgb_frame_w)" />
    	<param name="rgb_frame_h" value="$(arg rgb_frame_h)" />
    	
        <param name="rgb_frame_id" value="$(arg rgb_frame_id)" />
        <param name="depth_frame_id" value="$(arg depth_frame_id)" />
    
        <param name="depth_unit" value="$(arg depth_unit)" />
        <param name="depth_scale" value="$(arg depth_scale)" />
        
        <param name="depth_fx" value="$(arg depth_fx)" />
        <param name="depth_fy" value="$(arg depth_fx)" />
        
        <param name="depth_cx" value="$(arg depth_cx)" />
        <param name="depth_cy" value="$(arg depth_cy)" />
        
        <param name="depth_uv_enable_min" value="$(arg depth_uv_enable_min)" />
        <param name="depth_uv_enable_max" value="$(arg depth_uv_enable_max)" />
        
        <param name="topic_depth_points_id" value="$(arg topic_depth_points_id)" />
        <param name="topic_depth_registered_points_id" value="$(arg topic_depth_registered_points_id)" />
        
        <param name="topic_image_rgb_raw_id" value="$(arg topic_image_rgb_raw_id)" />
        <param name="topic_image_depth_raw_id" value="$(arg topic_image_depth_raw_id)" />

        <param name="topic_image_infrared_raw_id" value="$(arg topic_image_infrared_raw_id)" />
        
        <param name="debug_depth_unit" value="$(arg debug_depth_unit)" />
        
        <param name="rgb_camera_info_url" value="$(arg rgb_camera_info_url)" />
        <param name="ir_camera_info_url" value="$(arg ir_camera_info_url)" />

    </node>
    
    <include file="$(find realsense_camera)/launch/realsense_frames.launch">
        <arg name="camera" value="$(arg camera)" />
    </include>
</launch>
